lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/NetconfBaseServiceImpl.java

```diff
@@ -106,6 +107,9 @@ public NetconfBaseServiceImpl(final NodeId nodeId,
         Preconditions.checkArgument(!NETCONF_RUNNING_QNAME.equals(targetDatastore),
                 "Running datastore cannot be deleted.");
 
+        Preconditions.checkArgument(!Running.QNAME.equals(targetDatastore),
```

Comment:
why do we have two separate checks? how different are the two QNames?

Reply:
We should update preconditions for targetDatastore when calling `delete-config`. But as we discussed today, it should be rather called on _NetconfNmdaBaseServiceImpl_, because according to RFC6241 `delete-config` can't be called with Running datastore as target. _NetconfNmdaBaseService_ is extension to _NetconfBaseService_ in lighty, where are **RFC6241** methods like `edit-config`, `get-config` etc. implemented. Nmda implements two new methods from **RFC8526** for extended datastore types from **RFC 8342**, so as is understood by me _NetconfBaseService_ methods for edit, get, delete are applicable for older types of datastores used before new NMDA RFC and those new are used for the new ones. As the _NetconfNmdaBaseService_ extends _NetconfBaseService_, override of delete-config is probably the best choice to check if target datastore is not Running QName from the ietf-datastores and if it is, it's replaced by older Running QName and then called parent method, so expecting Preconditions check error.